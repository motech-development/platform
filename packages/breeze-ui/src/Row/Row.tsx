import { FC, memo, ReactNode } from 'react';
import styled from 'styled-components';

interface IRowContainer {
  $columns: number;
  $gutter: string;
}

const RowContainer = styled.div<IRowContainer>`
  ${({ $columns, $gutter }) => `
    display: grid;
    gap: ${$gutter};
    grid-template-columns: repeat(${$columns},1fr);
  `}
`;

export interface IRowProps {
  children: ReactNode;
  columns?: number;
  gutter?: string;
}

const Row: FC<IRowProps> = ({ children, columns = 12, gutter = '1rem' }) => (
  <RowContainer $columns={columns} $gutter={gutter}>
    {children}
  </RowContainer>
);

export default memo(Row);
