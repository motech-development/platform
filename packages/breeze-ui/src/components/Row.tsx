import { FC, ReactNode } from 'react';
import { TColumn, TGap } from '../utils/grid';

export interface IRowProps {
  children: ReactNode;
  columns?: TColumn;
  gutter?: TGap;
}

const Row: FC<IRowProps> = ({ children, columns = 12, gutter = 0 }) => (
  <>
    {/* @tailwind: grid-cols-1 grid-cols-2 grid-cols-3 grid-cols-4 grid-cols-5 grid-cols-6 grid-cols-7 grid-cols-8 grid-cols-9 grid-cols-10 grid-cols-11 grid-cols-12 */}
    {/* @tailwind: gap-0 gap-1 gap-2 gap-3 gap-4 gap-5 */}
    <div className={`grid grid-cols-${columns} gap-${gutter}`}>{children}</div>
  </>
);

export default Row;
