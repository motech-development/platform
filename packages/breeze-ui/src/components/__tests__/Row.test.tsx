import { render } from '@testing-library/react';
import Row from '../Row';

describe('Row', () => {
  it('should render with the default styles', () => {
    const { container } = render(
      <Row>
        <div data-testid="col" />
      </Row>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should output the correct styles with custom sizes', () => {
    const { container } = render(
      <Row columns={10} gutter={5}>
        <div data-testid="col" />
      </Row>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render content', () => {
    const { container } = render(
      <Row columns={10} gutter={5}>
        <div data-testid="col" />
      </Row>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
