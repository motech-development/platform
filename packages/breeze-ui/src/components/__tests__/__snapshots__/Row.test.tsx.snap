// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should output the correct styles with custom sizes 1`] = `
<div
  class="grid grid-cols-10 gap-5"
>
  <div
    data-testid="col"
  />
</div>
`;

exports[`Row should render content 1`] = `
<div
  class="grid grid-cols-10 gap-5"
>
  <div
    data-testid="col"
  />
</div>
`;

exports[`Row should render with the default styles 1`] = `
<div
  class="grid grid-cols-12 gap-0"
>
  <div
    data-testid="col"
  />
</div>
`;
